build: remove vendored Babel typings

petebacondarwin opened the pull request:
## Second attempt after previous PR was reverted (https://github.com/angular/angular/pull/33176)


These were getting included in the @angular/localize package.
Instead, patch the upstream files to work with TS typeRoots option

See bazelbuild/rules_nodejs#1033

alexeagle committed:
build: remove vendored Babel typings

These were getting included in the @angular/localize package.
Instead, patch the upstream files to work with TS typeRoots option

See bazelbuild/rules_nodejs#1033

gkalpak reviewed (approved):
LGTM for `dev-infra`.

petebacondarwin commented:
This seems to be passing on Windows now: https://circleci.com/workflow-run/c6ed6197-5f49-474c-ba43-4b8defbb01f7

petebacondarwin commented:
Caretaker:
This was reverted because it was failing on Windows.
I have fixed this by adding a few `typeRoot` properties to the failing tsconfig files.
I then checked this is passing on Windows by modifying the CircleCI config to trigger the windows tests, then I removed the CircleCI config change.
